### PR TITLE
Remove X-Requested-With header when performing CORS requests

### PR DESCRIPTION
**Pull request by neilius**
This PR prevents the `X-Requested-With` header from being set when making CORS requests, since that header causes an undesired preflight OPTIONS request. See #83 for original mention.

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: libs/util/http.client.js [hunk 1]
```diff
@@ -43,10 +43,11 @@ if (!String.prototype.trim) {
   };
 }
 
-function normalizeHeaders(headers, method) {
-    var normalized = {
-        'X-Requested-With': 'XMLHttpRequest'
-    };
+function normalizeHeaders(headers, method, isCors) {
+    var normalized = {};
+    if(!isCors) {
```
Comment (redonkulus) [inline libs/util/http.client.js]: add space after if

File: libs/util/http.client.js [hunk 2]
```diff
@@ -130,8 +131,8 @@ function mergeConfig(config) {
 function doXhr(method, url, headers, data, config, callback) {
     var options, timeout;
 
+    headers = normalizeHeaders(headers, method, config.cors);
```
Comment (redonkulus) [inline libs/util/http.client.js]: why was this moved up? do we not want it merged first?

Comment (neilius) [inline libs/util/http.client.js]: `mergeConfig` doesn't appear to actually merge the config passed in, it just returns a new config object using some of the properties from the passed-in config to initialize the new one. This means that the `cors` property is no longer present when `normalizeHeaders` is called. I moved it up so that I could access the `cors` property before `mergeConfig` wipes it out. Perhaps `mergeConfig` needs to be reworked a bit?

Comment (redonkulus) [inline libs/util/http.client.js]: got it.

**Comment by Vijar** [comment]
:ship: 

@neilius can you squash your commits into one?

**Comment by neilius** [comment]
@Vijar sure thing, it's now squashed

**Comment by Vijar** [comment]
Thanks!
